### ISSUE TITLE
Unsatisfactory reconstruction effect on Tank and Temples dataset

### ISSUE THREAD

**Issue by YuhsiHu** [alibaba/cascade-stereo#23]
Firstly, thank you for your great work and excellent code.
The pre-trained model perform perfectly on DTU dataset. However, pre-trained model cannot reconstruct on other dataset, such as Tank and Temples.
I resized the images in Tank and Temples from 1920X1080 to 1600X1200.
The param I set is: --dataset=general_eval --batch_size=1 --testpath=$TESTPATH  --testlist=$TESTLIST --loadckpt $CKPT_FILE --outdir $save_results_dir  --interval_scale 1.06 --max_h=2048 --max_w=2048$

The results are like this:
![image](https://user-images.githubusercontent.com/34529345/110742702-cb615100-8271-11eb-8c10-ed2914192255.png)

Could you please help me how can I get results in your paper?

**Comment by gxd1994** [comment]
hi，if you resize the image，the intrinsic matrix also need to rescale




------------------ Original ------------------

**Comment by YuhsiHu** [comment]
Hi, thank you for your answer!
This time I didn't change the size of the picture, I used the original size: 1920X1080.
Howerver, the result is slightly better than before, but it is still not as perfect as the paper shows.
![image](https://user-images.githubusercontent.com/34529345/111457536-45eb0e80-8753-11eb-9894-733438f2bafe.png)

> hi，if you resize the image，the intrinsic matrix also need to rescale
> […](#)
> ------------------ Original ------------------ From: YuhsiHu <notifications@github.com&gt; Date: 周四,3月 11,2021 13:59 To: alibaba/cascade-stereo <cascade-stereo@noreply.github.com&gt; Cc: Subscribed <subscribed@noreply.github.com&gt; Subject: Re: [alibaba/cascade-stereo] Unsatisfactory reconstruction effect on Tank and Temples dataset (#23) Firstly, thank you for your great work and excellent code. The pre-trained model perform perfectly on DTU dataset. However, pre-trained model cannot reconstruct on other dataset, such as Tank and Temples. I resized the images in Tank and Temples from 1920X1080 to 1600X1200. The param I set is: --dataset=general_eval --batch_size=1 --testpath=$TESTPATH --testlist=$TESTLIST --loadckpt $CKPT_FILE --outdir $save_results_dir --interval_scale 1.06 --max_h=2048 --max_w=2048$ The results are like this: Could you please help me how can I get results in your paper? — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by agenthong** [comment]
Just don't edit the codes.
Or use the original repo.

**Comment by YuhsiHu** [comment]
> Just don't edit the codes.
> Or use the original repo.

This is the result of the original code and the pre-trained model, which is slightly different from that in the paper. In addition, MVSNet will not have so many outstanding versions if it is only used as toys and nobody wants to improve it.

**Comment by agenthong** [comment]
> > Just don't edit the codes.
> > Or use the original repo.
> 
> This is the result of the original code and the pre-trained model, which is slightly different from that in the paper. In addition, MVSNet will not have so many outstanding versions if it is only used as toys and nobody wants to improve it.

I can reproduce the result by original codes.
![image](https://user-images.githubusercontent.com/59220627/113805179-a093eb00-9792-11eb-9f83-9814f9cc7a7b.png)

**Comment by YuhsiHu** [comment]
> > > Just don't edit the codes.
> > > Or use the original repo.
> > 
> > 
> > This is the result of the original code and the pre-trained model, which is slightly different from that in the paper. In addition, MVSNet will not have so many outstanding versions if it is only used as toys and nobody wants to improve it.
> 
> I can reproduce the result by original codes.
> ![image](https://user-images.githubusercontent.com/59220627/113805179-a093eb00-9792-11eb-9f83-9814f9cc7a7b.png)

it is good on DTU dataset but some results in Tank&Temple are not satisfactory.

**Comment by wangchengze001** [comment]
@wln19 hello，do you get good reconstruct in tanks？ i face the same question

**Comment by wangchengze001** [comment]
> Hi, thank you for your answer! This time I didn't change the size of the picture, I used the original size: 1920X1080. However, the result is slightly better than before, but it is still not as perfect as the paper shows. ![image](https://user-images.githubusercontent.com/34529345/111457536-45eb0e80-8753-11eb-9894-733438f2bafe.png)
> 
> > hi，if you resize the image，the intrinsic matrix also need to rescale
> > […](#)
> > ------------------ Original ------------------ From: YuhsiHu <[notifications@github.com](mailto:notifications@github.com)> Date: 周四,3月 11,2021 13:59 To: alibaba/cascade-stereo <[cascade-stereo@noreply.github.com](mailto:cascade-stereo@noreply.github.com)> Cc: Subscribed <[subscribed@noreply.github.com](mailto:subscribed@noreply.github.com)> Subject: Re: [alibaba/cascade-stereo] Unsatisfactory reconstruction effect on Tank and Temples dataset (#23) Firstly, thank you for your great work and excellent code. The pre-trained model perform perfectly on DTU dataset. However, pre-trained model cannot reconstruct on other dataset, such as Tank and Temples. I resized the images in Tank and Temples from 1920X1080 to 1600X1200. The param I set is: --dataset=general_eval --batch_size=1 --testpath=$TESTPATH --testlist=$TESTLIST --loadckpt $CKPT_FILE --outdir $save_results_dir --interval_scale 1.06 --max_h=2048 --max_w=2048$ The results are like this: Could you please help me how can I get results in your paper? — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe.

Hi, excuse me , do you solve the problem , i meet the same question

**Comment by YuhsiHu** [comment]
I think maybe the network is finetuned on Tanks&Temples dataset, or some params are different from configs on DTU.

**Comment by xiaomingHUST** [comment]
> I think maybe the network is finetuned on Tanks&Temples dataset, or some params are different from configs on DTU.

hi,when you test on the dataset tanks and temples,do you use the fusion method that offered.but in my case, the gipuma fusion method needs much more than 48G memory,it's strange.
could you please check the resources the fusion step needs.
thanks a lot.

**Comment by YuhsiHu** [comment]
> > I think maybe the network is finetuned on Tanks&Temples dataset, or some params are different from configs on DTU.
> 
> hi,when you test on the dataset tanks and temples,do you use the fusion method that offered.but in my case, the gipuma fusion method needs much more than 48G memory,it's strange. could you please check the resources the fusion step needs. thanks a lot.

Yes I used the fusion method that offered. Gipuma fusion method needs GPU and the image size of this dataset is large.

**Comment by xiaomingHUST** [comment]
> > > I think maybe the network is finetuned on Tanks&Temples dataset, or some params are different from configs on DTU.
> > 
> > 
> > hi,when you test on the dataset tanks and temples,do you use the fusion method that offered.but in my case, the gipuma fusion method needs much more than 48G memory,it's strange. could you please check the resources the fusion step needs. thanks a lot.
> 
> Yes I used the fusion method that offered. Gipuma fusion method needs GPU and the image size of this dataset is large.

thanks for your reply. and one more question, when you test on the tanks and temples dataset, which camera file do you use. 
i just download the preprocess data in MVSNet proj, but the range of my point model is strange. in other issue ,someone mentioned the the short depth range data, do you use that file and how to get this data? thanks again.

**Comment by YuhsiHu** [comment]
I download the dataset from original MVSNet repo. You can follow the instructions in their README.md

**Comment by xiaomingHUST** [comment]
> I download the dataset from original MVSNet repo. You can follow the instructions in their README.md

ok,I will have a try . thanks !

**Comment by zhao-you-fei** [comment]
@gxd1994 @YuhsiHu @xiaomingHUST @wangchengze001 @agenthong 
请问我按照casmvsnet流程测试T&T数据集出现以下效果该怎么有效解决,filter-method默认normal
最终融合:
![image](https://user-images.githubusercontent.com/75787742/196883830-09a5b218-3e9b-4517-b1ba-6cf278d7421f.png)
ply_local文件夹点云:
![image](https://user-images.githubusercontent.com/75787742/196884707-6d0e074e-d54e-4e3f-bd05-039df040fd1a.png)

深度图:
![image](https://user-images.githubusercontent.com/75787742/196884620-791aa77c-95c3-4820-bd63-82f9fef06587.png)
